internal/overlord/servstate/handlers.go

```diff
@@ -356,8 +363,12 @@ func (s *serviceData) startInternal() error {
 	done := make(chan struct{})
 	go func() {
 		waitErr := s.cmd.Wait()
+		childErr := s.waitProcessGroup()
```

Comment:
Nit: I think this should just be called `err`, as in `err := s.waitProcessGroup()`. (The `waitErr` is a bit of an anomaly as it's not tested right away but passed elsewhere.)

Reply:
Agree, adjusted.